packages/ricos-editor/web/src/tiptap/RicosEditor.tsx

```diff
@@ -65,11 +55,22 @@ class RicosEditor extends React.Component<Props, RicosEditorState> implements Ri
   }
 
   componentWillReceiveProps(newProps: Props) {
-    if (
-      newProps.injectedContent &&
-      !isEqual(this.props.injectedContent, newProps.injectedContent)
-    ) {
-      this.tiptapAdapter.setDraftContent(newProps.injectedContent);
+    if (newProps.injectedContent) {
+      const isRichContentSchema = newProps.schemaType === 'RichContent';
+      if (
+        isRichContentSchema
+          ? !isEqual(
+              (this.props.injectedContent as RichContent).nodes,
+              (newProps.injectedContent as RichContent).nodes
+            ) &&
+            !isEqual(
+              (this.props.injectedContent as RichContent).documentStyle,
+              (newProps.injectedContent as RichContent).documentStyle
+            )
+          : !isEqual(this.props.injectedContent, newProps.injectedContent)
+      ) {
+        this.props.schema?.setContent(newProps.injectedContent);
+      }
```

Comment:
Suggestion:
Maybe the next person who comes across this `if` statement doesn't waste time understanding.

And another thing, are you sure is `&&` and not `||`?

```suggestion
    if (newProps.injectedContent) {
      const isRichContentSchema = newProps.schemaType === 'RichContent';
      const isRichContentSchemaContentChanged = !isEqual(
              (this.props.injectedContent as RichContent).nodes,
              (newProps.injectedContent as RichContent).nodes;
      const isDocumentStyleChanged = !isEqual(
              (this.props.injectedContent as RichContent).documentStyle,
              (newProps.injectedContent as RichContent).documentStyle;
      const isDraftSchemaContentChanged =   !isEqual(this.props.injectedContent, newProps.injectedContent);
      if (
        isRichContentSchema
          ? isRichContentSchemaContentChanged && isDocumentStyleChanged
          : isDraftSchemaContentChanged
      ) {
        this.props.schema?.setContent(newProps.injectedContent);
      }
```

Reply:
@noamtwix 

> Suggestion:
> Maybe the next person who comes across this if statement doesn't waste time understanding.
Thanks, I'll refactor it :)

> And another thing, are you sure is && and not ||?
Great catch! thank you 🔥

Reply:
@maimonav 

No problem.
When it's easy to read it's easy to catch ;)